templates/terraform/resource.erb

```diff
@@ -357,7 +366,10 @@ func resource<%= resource_name -%>Update(d *schema.ResourceData, meta interface{
 <%  if object.input -%>
     d.Partial(true)
 
-<%  properties_by_custom_update(object.root_properties).each do |key, props| -%>
+<% properties_by_custom_update(object.root_properties)
+    .sort_by {|k, _| k.nil? "" : k[:update_id]}
```

Comment:
You can move this directly to `properties_by_custom_update`, adding ordering to the output is fine